syne_tune/optimizer/schedulers/searchers/bayesopt/datatypes/hp_ranges.py

```diff
@@ -65,12 +67,15 @@ def __init__(
         :param prefix_keys: If given, these keys into `config_space` come first
             in the internal ordering, which determines the internal
             encoding
+        :param strictly_non_binary_choice_hyperparameter_keys: If given, these keys are considered to be strictly
```

Comment:
I don't understand the rationale for this. The binary vs non-binary encoding is an internal property of the child class. Nothing in warmstarting should depend on how categorical HPs are internally encoded.

Reply:
Here is the part where we depend on the dimension of each hyperparameter: https://github.com/awslabs/syne-tune/blob/main/syne_tune/optimizer/schedulers/searchers/utils/warmstarting.py#L105-L108

Since binary hyperparameters have dimensionality of 1, we calculate size of this kernel wrong.

Reply:
Better solution: #241